DailyKitTests/CallParticipantsTests.swift

```diff
@@ -143,3 +147,118 @@ final class CallParticipantsTests: XCTestCase, TestSupport {
+        let previous = makeCallParticipants(visible: [
+            CallParticipant(id: 0, username: "0", hasVideo: false),
+            CallParticipant(id: 1, username: "1", hasVideo: false),
+            CallParticipant(id: 2, username: "2", hasVideo: false),
+        ])
+        let current = makeCallParticipants(visible: [
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+        ])
+
+        let participants = previous.stabilize(current)
+
+        assertParticipants(participants, visibleEquals: [
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+        ])
+    }
+
+    func testStabilizePreservesNewValuesWhenOthersIsBigger() throws {
+        let previous = makeCallParticipants(visible: [
+            CallParticipant(id: 0, username: "0", hasVideo: false),
+            CallParticipant(id: 1, username: "1", hasVideo: false),
+            CallParticipant(id: 2, username: "2", hasVideo: false),
+        ])
+        let current = makeCallParticipants(visible: [
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+            CallParticipant(id: 4, username: "4", hasVideo: true),
+            CallParticipant(id: 0, username: "0", hasVideo: true),
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+        ])
+
+        let participants = previous.stabilize(current)
+
+        assertParticipants(participants, visibleEquals: [
+            CallParticipant(id: 0, username: "0", hasVideo: true),
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+            CallParticipant(id: 4, username: "4", hasVideo: true),
+        ])
+    }
+
+    func testStabilizePreservesNewValuesWhenOthersIsSmaller() throws {
+        let previous = makeCallParticipants(visible: [
+            CallParticipant(id: 0, username: "0", hasVideo: true),
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+            CallParticipant(id: 4, username: "4", hasVideo: true),
+        ])
+        let current = makeCallParticipants(visible: [
+            CallParticipant(id: 1, username: "1", hasVideo: true),
+            CallParticipant(id: 2, username: "2", hasVideo: true),
+            CallParticipant(id: 3, username: "3", hasVideo: true),
+        ])
+
+        let participants = previous.stabilize(current)
+
+        assertParticipants(participants, visibleEquals: [
```

Comment:
The `CallParticipants` values in this file are created directly via an initializer as opposed to via `CallParticipants.Builder`, so the prioritization rules are not used, and the key/index in the `visible` dictionary is what makes one participant higher than another. (stable visible order between two collection values vs priority order within a single collection value)

All of the participants in `previous` here should have had `hasVideo: false`, but that shouldn't affect the order. I've updated that, but I'm questioning using `hasVideo` here. Would using a different `username` instead of `hasVideo` value avoid such ambiguity? I think the difference between the tests in this file and those in `CallParticipantsBuilderTests` is not obvious.

Reply:
Thanks, for elaborating! Tbh I was still a little confused as to what made `3` move up in the order vs keeping the original, but after reading `stabilize()` itself I _think_ I get it. 

Re the ambiguity - I think it wasn't so much that which confused me, but not understanding the intended behavior of `stabilize()`. I'm going to run this later for my own sake to make sure I get it now, but this lgtm!